scripts/secnetperf.ps1

```diff
@@ -225,19 +232,29 @@ if ($isWindows) {
 Configure-DumpCollection $Session
 
 # Install any dependent drivers.
-if ($useXDP) { Install-XDP $Session $RemoteDir }
+if ($useXDP -and $isWindows) { Install-XDP $Session $RemoteDir }
 if ($io -eq "wsk") { Install-Kernel $Session $RemoteDir $SecNetPerfDir }
 
 if (!$isWindows) {
     # Make sure the secnetperf binary is executable.
     Write-Host "Updating secnetperf permissions"
+    $GRO = "on"
+    $ENABLE_XDP=0
+    if ($io -eq "xdp") {
+        $GRO = "off"
```

Comment:
Why do you need to disable GRO to use XDP?

Reply:
Decryption in connection.c start failing with XDP & GRO.
Maybe need additional implementation to split the merged chunks?

Reply:
GRO might not work with generic XDP? (SKB_MODE)
<img width="691" alt="image" src="https://github.com/microsoft/msquic/assets/5763034/629e483d-5183-4d34-8556-8abe79d467a2">
https://conferences.sigcomm.org/sigcomm/2023/files/workshop-ebpf/1-CloudNative.pdf

and LRO (HW_GRO) does not work with XDP?
https://www.infradead.org/~mchehab/kernel_docs/networking/device_drivers/microsoft/netvsc.html

Reply:
might be supported? need investigation. kernel version, library version, host/guest driver
https://lpc.events/event/11/contributions/939/attachments/771/1551/xdp-multi-buff.pdf

Reply:
I didn't see you disable GRO for all the other tests. Also, can't you programmatically disable it for your XDP socket instead of doing it system-wide?

Reply:
I realized that GRO in duonic is default off. And duonic support DRV mode attach.
DRV (driver) mode can reside with GRO because packet redirection by XDP program is "before" GRO.
DRV mode XDP doesn't GRO actually.
I tried 4 combination with duonic, SKB_MODE/DRV_MODE and GRO on/off. However the maximum ethernet frame size is always 1514. I don't actually see GROed packet. I saw larger packets on eth0 on Azure VM with GRO on then decryption failure.

As far as I checked ethtool code and document, it opens another socket by `socket(AF_NETLINK, SOCK_RAW, NETLINK_GENERIC);`. Then send payload, which is not well documented though, to kernel to set/unset flags. Document says that netlink feature is kind of ioctl so that netlink operation should be for each device (eth0 on Azure VM).
This socket (AF_NETLINK) is not associated with XDP socket. I believe it is difficult to disable/enable for each socket level. It is technically possible to do programmatically for system-wide, but it is kind of re-inventing the wheel of ethtool in msquic